datumaro/plugins/imagenet_format.py

```diff
@@ -0,0 +1,140 @@
+            labels = self._parse_labels(labels)
+        elif isinstance(labels, list):
+            assert all(isinstance(lbl, str) for lbl in labels), labels
+        else:
+            raise TypeError("Unexpected type of 'labels' argument: %s" % labels)
+
+        self._categories = self._load_categories(labels)
+        self._items = list(self._load_items(path).values())
+
+    @staticmethod
+    def _parse_labels(path):
+        with open(path, encoding='utf-8') as labels_file:
+            return [s.strip() for s in labels_file]
+
+    def _load_categories(self, labels):
+        label_cat = LabelCategories()
+        for label in labels:
+            label_cat.add(label)
+
+        return { AnnotationType.label: label_cat }
+
+    def _load_items(self, path):
+        items = defaultdict(dict)
+
+        with open(path, encoding='utf-8') as f:
+            for line in f:
+                image_name, label_id = line.split()
+                items[image_name[:-4]] = DatasetItem(
+                    id=image_name[:-4], subset=self._subset,
+                    image=image_name,
+                    annotations=[Label(label=label_id)])
+        return items
+
+
+class ImagenetImporter(Importer):
+
+    @classmethod
+    def detect(cls, path):
+        with logging_disabled(log.WARN):
+            return (len(cls.find_sources(path)) != 0)
+
+    def __call__(self, path, **extra_params):
+        from datumaro.components.project import Project # cyclic import
+        project = Project()
+
+        subsets = self.find_sources(path)
+
+        if len(subsets) == 0:
+            raise Exception("Failed to find 'imagenet' dataset at '%s'" % path)
+
+        for ann_name, ann_file in subsets.items():
+            project.add_source(ann_name, {
+                    'url': ann_file,
+                    'format': 'imagenet',
+                    'options': dict(extra_params),
+                })
+
+        return project
+
+    @staticmethod
+    def find_sources(path):
+        subsets = defaultdict(dict)
+        subset_paths = [p for p in glob(osp.join(path, '*.txt'))
+                if 'synsets' not in osp.basename(p)]
+
+        for subset_path in subset_paths:
+            subset_name = osp.splitext(osp.basename(subset_path))[0]
+            subsets[subset_name] = subset_path
+        return subsets
+
+
+class ImagenetConverter(Converter):
+    DEFAULT_IMAGE_EXT = '.jpg'
+
+    def apply(self):
+        subset_dir = self._save_dir
+        extractor = self._extractor
+        images_dir = osp.join(subset_dir, ImagenetPath.IMAGES_DIR)
+        os.makedirs(images_dir, exist_ok=True)
+        self._images_dir = images_dir
+        image_labels = defaultdict(dict)
+        for subset_name, subset in self._extractor.subsets().items():
+            annotation_file = osp.join(subset_dir, '%s.txt' % subset_name)
+            annotation = ''
+            for item in subset:
+                label = item.annotations[0].label
```

Comment:
This line can easily throw an error, if an item does not have annotations. I also think we should cover the case with multiple labels per image - both for reading and saving.